components/tabs/__tests__/index.test.tsx

```diff
@@ -50,73 +42,169 @@ describe('Tabs', () => {
         </Tabs.Item>
       </Tabs>,
     )
-    expect(errorSpy).toHaveBeenCalled()
-    errorSpy.mockRestore()
+
+    wrapper.find('header').find('.tab').at(1).simulate('click', nativeEvent)
+    expect(changeHandler).not.toHaveBeenCalled()
+  })
+
+  it('should return default nav', () => {
+    nav({})
   })
 
-  it('should use label as key when value is missing', async () => {
-    const errorSpy = jest.spyOn(console, 'error').mockImplementation(() => {})
+  it('should ingnore click in uncontroled component', () => {
     const wrapper = mount(
-      <Tabs>
-        <Tabs.Item label="label1">test-1</Tabs.Item>
-        <Tabs.Item label="label2">test-2</Tabs.Item>
+      <Tabs initialValue="1" value="2" showDivider={true} varient="solid">
+        <Tabs.Item label="label1" value="1">
+          1
+        </Tabs.Item>
+        <Tabs.Item label="label2" value="2" disabled>
+          2
+        </Tabs.Item>
       </Tabs>,
     )
-    expect(errorSpy).not.toHaveBeenCalled()
+    wrapper.find('header').find('.tab').at(0).simulate('click', nativeEvent)
+  })
+  it('should be able to update tab-item props', () => {
+    function App() {
+      const [label1, setLabel1] = useState('label1')
+      useEffect(() => {
+        setLabel1('label11')
+      }, [])
+      return (
+        <Tabs initialValue="1" value="2" showDivider={true}>
+          <Tabs.Item label={label1} value="1">
+            1
+          </Tabs.Item>
+          <Tabs.Item label="label2" value="2" disabled>
+            2
+          </Tabs.Item>
+        </Tabs>
+      )
+    }
+    mount(<App />)
+  })
+})
+
+describe('useImperative', () => {
+  it('should work with useImperative', () => {
+    const { useImperative } = Tabs
+    function App() {
+      const { ref, currentTab } = useImperative()
+      useEffect(() => {
+        currentTab()
+        currentTab('2')
+      }, [])
+      return (
+        <Tabs ref={ref}>
+          <Tabs.Item label="label1" value="1">
+            1
+          </Tabs.Item>
+          <Tabs.Item label="label2" value="2" disabled>
+            2
+          </Tabs.Item>
+        </Tabs>
+      )
+    }
+
+    mount(<App />)
+  })
+})
+
+describe('utils', () => {
+  it('should cover a unreachable brach due to a ts bug', () => {
+    reduceStatus({})
+  })
+
+  it('should return corrent colors with virant and status combination', () => {
+    const status = ['disabled', 'active', 'hover', 'default']
+    const varients = ['line', 'solid']
+    status.forEach(s => {
+      varients.forEach(v => {
+        defaultGetColor(palette, v, s)
+      })
+    })
+  })
+})
 
-    wrapper.setProps({ value: 'label2' })
-    await updateWrapper(wrapper, 350)
-    const active = wrapper.find('header').find('.active')
-    expect(active.text()).toContain('label2')
-    errorSpy.mockRestore()
+describe('useFixedWidth', () => {
+  it('should work on a null component', () => {
+    function App() {
+      const [width, ref] = useFixedWidth<HTMLDivElement>(null)
```

Comment:
width is not used here

Reply:
Jest can not get `width` correctly

Reply:
https://github.com/jsdom/jsdom/issues/135